gammapy/datasets/spectrum.py

```diff
@@ -326,14 +326,23 @@ def read(cls, filename, format="ogip", checksum=False, name=None, **kwargs):
         from .io import OGIPDatasetReader
 
         if format == "gadf":
-            return super().read(
+            mapd = super().read(
                 filename, format="gadf", checksum=checksum, name=name, **kwargs
             )
+            if (
+                mapd.counts is None
+                and mapd.background is None
+                and mapd.exposure is None
+                and mapd.edisp is None
+            ):
```

Comment:
in which case does it happen ? is there a way to fix it ?

Reply:
Well, it is a perhaps very dirty way of spotting that this dataset is indeed saved as ogip, and if you try to force-read it as gadf it will return a dataset with everything set to None. Probably doing all those ands is a bit of an overkill.

Reply:
Maybe the default format should be`None` and it would be guessed based on this such as we use `ogip` if the output of `gadf` is all `None`. 
But if possible the format should be guessed via some mandatory keys existing in the gadf or ogip headers.

Reply:
Both done. Defaults to None (guess format), and guess the format based on extensions 

```if "SPECTRUM" in hdulist and "OGIP" in hdulist[1].header["HDUCLASS"]``` ---> OGIP
```elif ("COUNTS" in hdulist and "EXPOSURE" in hdulist and "EDISP" in hdulist):``` ---> GADF

raises an error now if it is a MapDataset rather than Spectrum dataset ('IMAGE' in hdulist[1].header["XTENSION"])